cmd/contour/serve.go

```diff
@@ -210,37 +221,46 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	// wrap eventRecorder in a converter for objects from the dynamic client.
 	dynamicHandler := &k8s.DynamicClientHandler{
 		Next:      eventRecorder,
-		Converter: k8s.NewUnstructuredConverter(),
+		Converter: converter,
```

Comment:
Why not pass the ctx.UseServiceAPITypes flag into k8s.NewUnstructuredConverter ?

Reply:
further, why not always register the service api types, as long as they aren't hooked up the API via an informer they can't hurt anything.

Reply:
The contract for `AddToScheme` (the kubebuilder way of adding things to a scheme) is different to `AddKnownTypes` and can return an error, we need to be able to bubble that up, sadly. So I had to add an error value to k8s.NewUnstructureConverter.